### PR TITLE
Chore: command's endpoints

### PR DESCRIPTION
**Pull request by ggazzo**

### CODE REVIEW

File: apps/meteor/app/slashcommand-asciiarts/lib/shrug.ts [hunk 1]
```diff
@@ -7,13 +6,15 @@ import { slashCommands } from '../../utils/lib/slashCommand';
...
-	Meteor.call('sendMessage', msg);
-}
-
-slashCommands.add('shrug', Shrug, {
-	description: 'Slash_Shrug_Description',
-	params: 'your_message_optional',
-});
+slashCommands.add(
+	'shrug',
+	(_command: 'shrug', params, item): void => {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommand-asciiarts/lib/shrug.ts:11]: An anonymous function is harder to debug.

File: apps/meteor/app/slashcommand-asciiarts/lib/tableflip.ts [hunk 2]
```diff
@@ -7,13 +6,15 @@ import { slashCommands } from '../../utils/lib/slashCommand';
...
-	Meteor.call('sendMessage', msg);
-}
-
-slashCommands.add('tableflip', Tableflip, {
-	description: 'Slash_Tableflip_Description',
-	params: 'your_message_optional',
-});
+slashCommands.add(
+	'tableflip',
+	(_command, params, item): void => {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommand-asciiarts/lib/tableflip.ts:11]: An anonymous function is harder to debug.

File: apps/meteor/app/slashcommands-archiveroom/server/server.ts [hunk 3]
```diff
@@ -1,71 +1,72 @@
...
 import { Rooms, Messages } from '../../models/server';
 import { slashCommands } from '../../utils/lib/slashCommand';
 import { api } from '../../../server/sdk/api';
 import { settings } from '../../settings/server';
 
-function Archive(_command: 'archive', params: string, item: IMessage): void {
-	let channel = params.trim();
+slashCommands.add(
+	'archive',
+	function Archive(_command, params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-archiveroom/server/server.ts:11]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-create/server/server.ts [hunk 4]
```diff
@@ -1,61 +1,62 @@
...
-	function getParams(str: string): string[] {
-		const regex = /(--(\w+))+/g;
-		const result = [];
-		let m;
-		while ((m = regex.exec(str)) !== null) {
-			if (m.index === regex.lastIndex) {
-				regex.lastIndex++;
+slashCommands.add(
+	'create',
+	function Create(_command: 'create', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-create/server/server.ts:11]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-create/server/server.ts [hunk 5]
```diff
@@ -1,61 +1,62 @@
...
+		function getParams(str: string): string[] {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-create/server/server.ts:12]: `nitpick` Maybe move it to the outer scope.

File: apps/meteor/app/slashcommands-help/server/server.ts [hunk 6]
```diff
@@ -17,55 +16,57 @@ interface IHelpCommand {
 	command: string;
 }
 
-function Help(_command: 'help', _params: string, item: IMessage): void {
-	const userId = Meteor.userId() as string;
-	const user = Users.findOneById(userId);
+slashCommands.add(
+	'help',
+	function Help(_command, _params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-help/server/server.ts:21]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-hide/server/hide.ts [hunk 7]
```diff
@@ -12,63 +11,65 @@ import { api } from '../../../server/sdk/api';
...
 
-function Hide(_command: 'hide', param: string, item: IMessage): void {
-	const room = param.trim();
-	const userId = Meteor.userId();
-	if (!userId) {
-		return;
-	}
+slashCommands.add(
+	'hide',
+	(_command: 'hide', param, item): void => {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-hide/server/hide.ts:16]: An anonymous function is harder to debug.

File: apps/meteor/app/slashcommands-invite/server/server.ts [hunk 8]
```diff
@@ -11,74 +10,75 @@ import { api } from '../../../server/sdk/api';
...
-	}
-	const usersFiltered = users.fetch().filter(function (user) {
-		const subscription = Subscriptions.findOneByRoomIdAndUserId(item.rid, user._id, {
-			fields: { _id: 1 },
-		});
-		if (subscription == null) {
-			return true;
+slashCommands.add(
+	'invite',
+	(_command: 'invite', params, item): void => {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-invite/server/server.ts:15]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-join/server/server.ts [hunk 9]
```diff
@@ -1,53 +1,54 @@
...
-	if (!room) {
-		api.broadcast('notify.ephemeralMessage', userId, item.rid, {
-			msg: TAPi18n.__('Channel_doesnt_exist', {
-				postProcess: 'sprintf',
-				sprintf: [channel],
-				lng: settings.get('Language') || 'en',
-			}),
+slashCommands.add(
+	'join',
+	(_command: 'join', params, item): void => {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-join/server/server.ts:11]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-leave/server/leave.ts [hunk 10]
```diff
@@ -11,7 +11,7 @@ import { Users } from '../../models/server';
  * Leave is a named function that will replace /leave commands
  * @param {Object} message - The message object
  */
-function Leave(_command: string, _params: string, item: IMessage): void {
+const Leave: SlashCommand<'leave'>['callback'] = function Leave(_command, _params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-leave/server/leave.ts:14]: Redundant type function declaration.

File: apps/meteor/app/slashcommands-me/server/me.ts [hunk 11]
```diff
@@ -10,7 +9,7 @@ import { slashCommands } from '../../utils/lib/slashCommand';
  */
 slashCommands.add(
 	'me',
-	function Me(_command: 'me', params: string, item: IMessage): void {
+	function Me(_command: 'me', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-me/server/me.ts:12]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-msg/server/server.ts [hunk 12]
```diff
@@ -12,42 +11,44 @@ import { api } from '../../../server/sdk/api';
...
-}
-
-slashCommands.add('msg', Msg, {
-	description: 'Direct_message_someone',
-	params: '@username <message>',
-	permission: 'create-d',
-});
+slashCommands.add(
+	'msg',
+	function Msg(_command: 'msg', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-msg/server/server.ts:16]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-mute/server/mute.ts [hunk 13]
```diff
@@ -11,44 +10,46 @@ import { api } from '../../../server/sdk/api';
...
  */
 
-function Mute(_command: 'mute', params: string, item: IMessage): void {
-	const username = params.trim().replace('@', '');
-	if (username === '') {
-		return;
-	}
+slashCommands.add(
+	'mute',
+	function Mute(_command: 'mute', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-mute/server/mute.ts:15]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-mute/server/unmute.ts [hunk 14]
```diff
@@ -11,43 +10,45 @@ import { api } from '../../../server/sdk/api';
...
-			msg: TAPi18n.__('Username_doesnt_exist', {
-				postProcess: 'sprintf',
-				sprintf: [username],
-				lng: settings.get('Language') || 'en',
-			}),
-		});
-	}
+slashCommands.add(
+	'unmute',
+	function Unmute(_command, params, item): void | Promise<void> {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-mute/server/unmute.ts:15]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-open/client/client.ts [hunk 15]
```diff
@@ -1,46 +1,47 @@
...
 import { Subscriptions, ChatSubscription } from '../../models/client';
 
-function Open(_command: 'open', params: string, _item: IMessage): void {
-	const dict: Record<string, string[]> = {
-		'#': ['c', 'p'],
-		'@': ['d'],
-	};
+slashCommands.add(
+	'open',
+	function Open(_command, params): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-open/client/client.ts:10]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-status/client/status.ts [hunk 16]
```diff
@@ -1,26 +1,27 @@
...
 import { slashCommands } from '../../utils/lib/slashCommand';
 import { settings } from '../../settings/server';
 import { api } from '../../../server/sdk/api';
 import { handleError } from '../../../client/lib/utils/handleError';
 
-function Status(_command: 'status', params: string, item: IMessage): void {
-	const userId = Meteor.userId() as string;
+slashCommands.add(
+	'status',
+	function Status(_command, params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-status/client/status.ts:11]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-status/server/status.ts [hunk 17]
```diff
@@ -1,36 +1,37 @@
...
 import { slashCommands } from '../../utils/lib/slashCommand';
 import { settings } from '../../settings/server';
 import { api } from '../../../server/sdk/api';
 import { Users } from '../../models/server';
 
-function Status(_command: 'status', params: string, item: IMessage): void {
-	const userId = Meteor.userId() as string;
+slashCommands.add(
+	'status',
+	function Status(_command: 'status', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-status/server/status.ts:11]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-topic/client/topic.ts [hunk 18]
```diff
@@ -1,31 +1,32 @@
...
 
-function Topic(_command: 'topic', params: string, item: IMessage): void {
-	if (Meteor.isClient && hasPermission('edit-room', item.rid)) {
-		Meteor.call('saveRoomSettings', item.rid, 'roomTopic', params, (err: Meteor.Error) => {
-			if (err) {
-				if (Meteor.isClient) {
-					handleError(err);
+slashCommands.add(
+	'topic',
+	function Topic(_command: 'topic', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-topic/client/topic.ts:11]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-topic/server/topic.ts [hunk 19]
```diff
@@ -1,21 +1,22 @@
...
-}
-
-slashCommands.add('topic', Topic, {
-	description: 'Slash_Topic_Description',
-	params: 'Slash_Topic_Params',
-	permission: 'edit-room',
-});
+slashCommands.add(
+	'topic',
+	function Topic(_command: 'topic', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-topic/server/topic.ts:8]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

File: apps/meteor/app/slashcommands-unarchiveroom/server/server.ts [hunk 20]
```diff
@@ -9,61 +8,63 @@ import { api } from '../../../server/sdk/api';
 import { roomCoordinator } from '../../../server/lib/rooms/roomCoordinator';
 import { RoomMemberActions } from '../../../definition/IRoomTypeConfig';
 
-function Unarchive(_command: 'unarchive', params: string, item: IMessage): void {
-	let channel = params.trim();
-	let room;
+slashCommands.add(
+	'unarchive',
+	function Unarchive(_command: 'unarchive', params, item): void {
```
Comment (tassoevan) [inline apps/meteor/app/slashcommands-unarchiveroom/server/server.ts:13]: Nesting this function declaration as a parameter expression damages readability, as evidenced by the indentation increase.

**Review by tassoevan — CHANGES_REQUESTED** [review]

**Review by juliajforesti — APPROVED** [review]

**Comment by ggazzo** [comment]
if I do it the other way, it will force me to type the functions, which is a dynamic type, and it looks horrible @tassoevan

**Pull request merged**